### PR TITLE
test(ui): Convert issues list tests to typescript

### PR DESCRIPTION
**Pull request by scttcper**

### CODE REVIEW

File: static/app/views/issueList/overview.spec.tsx [hunk 1]
```diff
@@ -57,12 +57,19 @@ const routerProps = {
 describe('IssueList', function () {
   let props;
```
Comment (scttcper) [inline static/app/views/issueList/overview.spec.tsx:58]: this is typed as any and why this test didn't take much to convert

File: static/app/__mocks__/api.tsx [hunk 2]
```diff
@@ -49,6 +49,7 @@ interface ResponseType extends ApiNamespace.ResponseMeta {
    * This will override `MockApiClient.asyncDelay` for this request.
    */
   asyncDelay?: AsyncDelay;
+  query?: Record<string, string | number | boolean | string[] | number[]>;
```
Comment (evanpurkhiser) [inline static/app/__mocks__/api.tsx:52]: damn, how many of these did I remove LOL

File: static/app/views/issueList/overview.spec.tsx [hunk 3]
```diff
@@ -172,16 +165,16 @@ describe('IssueList', function () {
...
-    let savedSearchesRequest;
-    let recentSearchesRequest;
-    let issuesRequest;
+    let savedSearchesRequest: jest.Mock;
+    let recentSearchesRequest: jest.Mock;
+    let issuesRequest: jest.Mock;
 
     beforeEach(function () {
-      StreamGroup.mockClear();
+      jest.mocked(StreamGroup).mockClear();
```
Comment (evanpurkhiser) [inline static/app/views/issueList/overview.spec.tsx:177]: 👍

**Review by evanpurkhiser — APPROVED** [review]

File: static/app/views/issueList/overview.spec.tsx (hunk 1 above)
Comment (evanpurkhiser) [reply static/app/views/issueList/overview.spec.tsx:58]: should we fix it 😬

**Pull request merged**